dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/constant/DefaultConstantService.java

```diff
@@ -54,12 +60,33 @@
 
     private IdentifiableObjectStore<Constant> constantStore;
 
+    private final CacheProvider cacheProvider;
+
+    private Cache<List> ALL_CONSTANTS_CACHE;
+
+    private static final String ALL_CONSTANTS_KEY = "x";
+
     public DefaultConstantService(
-        @Qualifier( "org.hisp.dhis.constant.ConstantStore" ) IdentifiableObjectStore<Constant> constantStore )
+        @Qualifier( "org.hisp.dhis.constant.ConstantStore" ) IdentifiableObjectStore<Constant> constantStore,
+        CacheProvider cacheProvider )
     {
         checkNotNull( constantStore );
+        checkNotNull( cacheProvider );
 
         this.constantStore = constantStore;
+        this.cacheProvider = cacheProvider;
+    }
+
+    @PostConstruct
+    public void init()
+    {
+        ALL_CONSTANTS_CACHE = cacheProvider.newCacheBuilder( List.class )
```

Comment:
Is there a reason to not initialise this field in the constructor?

Reply:
I was following the pattern for using cacheProvider elsewhere in DHIS2, e.g. DefaultAttributeService, DefaultIdentifiableObjectManager, DefaultOrganisationUnitService, etc.

Reply:
@larshelge do you know why we use post construct instead of constructor initialisation for caches?

Reply:
Simply because we used @ Autowired at the time @jbee :-) We should change over to constructor injection.

This should be done consistently across the system. Feel free to open a TECH issue for this so that we can track it and get it done.